pkg/validate/output.go

```diff
@@ -0,0 +1,38 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package validate
+
+import (
+	"context"
+
+	build "github.com/shipwright-io/build/pkg/apis/build/v1beta1"
+	"k8s.io/utils/pointer"
+)
+
+// BuildSpecOutputValidator implements validation interface to add validations for `build.spec.output`.
+type BuildSpecOutputValidator struct {
+	Build *build.Build // build instance for analysis
+}
+
+var _ BuildPath = &BuildSpecOutputValidator{}
+
+func (b *BuildSpecOutputValidator) ValidatePath(_ context.Context) error {
+	if b.Build.Spec.Output.Timestamp != nil {
+		if *b.Build.Spec.Output.Timestamp == "SourceTimestamp" {
+			if b.isSourceEmpty() {
```

Comment:
what about the combination of source type local and `SourceTimestamp` ?

Reply:
The Local code was not part of the previous PR to obtain the current source timestamp. I wanted to focus on the main ones Git and OCI first. The Local one would come in another PR later.

Reply:
where will Local fits? it applies as well for the `SourceTimestamp` type?

Reply:
Local should support all of the four styles, including SourceTimestamp, once I fixed that the Local path is producing a source timestamp to be used.